db/hints/sync_point.cc

```diff
@@ -135,13 +155,17 @@ sstring sync_point::encode() const {
     seastar::measuring_output_stream measure;
     ser::serializer<sync_point_v1>::write(measure, v1);
 
-    // Reserve version_size bytes for the version
-    sstring serialized{sstring::initialized_later{}, version_size + measure.size()};
-    seastar::simple_memory_output_stream out{serialized.data(), version_size + measure.size()};
-    ser::serializer<uint8_t>::write(out, 1);
+    // Reserve version_size bytes for the version and checksum_size bytes for the checksum
+    sstring serialized{sstring::initialized_later{}, version_size + measure.size() + checksum_size};
+
+    // Encode using V2 format
+    seastar::simple_memory_output_stream out{serialized.data(), version_size + measure.size() + checksum_size};
```

Comment:
How about `{serialized.data(), serialized.size()}`? The `serialized` string already has the correct length.